test-env/index.ts

```diff
@@ -193,10 +209,31 @@ const webDriverHost = process.env.BROWSERSTACK
 const testEnv: TestEnvironment = {
   session: new Session('default', 'default'),
   headless: !!process.env.HEADLESS,
+  setInitialWindowRectangle(rect: WindowRect) {
+    initialWindowRect = rect;
+  },
   ...(testEnvironments.find(
     ({ browser, driver }) =>
       browser === process.env.BROWSER && driver.host === webDriverHost
   ) || throwNoBrowserEnvironmentVariableError())
 };
 
+export async function getTestEnv(context?: Context): Promise<TestEnvironment> {
+  const { session } = testEnv;
+  const testAppPort = process.env.TEST_APP_PORT;
+
+  testEnv.testName = context?.test
+    ?.titlePath()
+    .join('.')
+    .replace(/ /g, '_');
+
+  await session.refresh();
+  await session.navigateTo(
+    `http://localhost:${testAppPort}/#${testEnv.testName}`
```

Comment:
Now we add the hash of test name to the URL so it's much easier to track which test is currently running.

Reply:
makes sense.